service/src/main/java/bio/terra/externalcreds/dataAccess/LinkedAccountDAO.java

```diff
@@ -71,6 +87,17 @@ public boolean deleteLinkedAccountIfExists(String userId, String providerId) {
     return jdbcTemplate.update(query, namedParameters) > 0;
   }
 
+  public boolean updateLinkAuthenticationStatus(int linkedAccountId, boolean isAuthenticated) {
+    var query =
+        "UPDATE linked_account SET is_authenticated = :isAuthenticated WHERE id = :linkedAccountId";
+    var namedParameters =
+        new MapSqlParameterSource()
+            .addValue("linkedAccountId", linkedAccountId)
+            .addValue("isAuthenticated", isAuthenticated);
+
+    return jdbcTemplate.update(query, namedParameters) > 0;
+  }
+
```

Comment:
I am torn about this function. On the one hand, this does just what we want. On the other hand, `upsertLinkedAccount` can do this too so this is extra code to maintain.

Reply:
Good point - I think using `upsertLinkedAccount` would be simpler. I just added `upsertLinkedAccount` to our `LinkedAccountService` (it was already in the DAO) and got rid of `updateLinkAuthenticationStatus`. That simplified a lot and also made it easier to address the comment you made above.